src/modules/entities/user/controllers/user.controller.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  ParseIntPipe,
+  Post,
+  Put,
+  Request,
+  UseGuards,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import { UserService } from '../services/user.service';
+import { CreateUserDto } from '../dto/user.create.dto';
+import { ApiBearerAuth, ApiResponse, ApiTags, OmitType } from '@nestjs/swagger';
+import { UpdateUserDto } from '../dto/user.update.dto';
+import { User } from '../entity/user.entity';
+import { JwtAuthGuard } from '../../../security/auth/guards/jwt.guard';
+import { UserResetPasswordDto } from '../dto/user.reset-password.dto';
+
+@ApiTags('User')
+@Controller('users')
+export class UserController {
+  constructor(private readonly userService: UserService) {}
+
+  @UsePipes(
+    new ValidationPipe({
+      transform: true,
+    }),
+  )
+  @Post()
+  @HttpCode(HttpStatus.CREATED)
+  async create(@Body() createUserDto: CreateUserDto) {
+    return this.userService.create(createUserDto);
+  }
+
+  @ApiResponse({
+    type: OmitType(User, ['password_hash', 'password_salt']),
+  })
+  @Get(':id')
+  @HttpCode(HttpStatus.OK)
+  async findOne(@Param('id', ParseIntPipe) id: number) {
+    return this.userService.findOneById(id);
+  }
+
+  @ApiResponse({
+    type: OmitType(User, ['password_hash', 'password_salt']),
```

Comment:
This is better you prevent from omitting sensitive fields on tyeorm query, and also use DTOs for responses